kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/data/DataServiceImpl.java

```diff
@@ -593,6 +618,15 @@ public void run() {
         });
     }
 
+    private synchronized void tryPublishInternal(DataMessage message) throws KuraException {
+        if (this.throttle.getToken()) {
```

Comment:
It should just be ``this.throttle.waitToken()``. Otherwise if this method is called from the publishing task and the DB is full and you don't have tokens, the CPU will spin doing nothing.

With waitToken() at least the thread will block until a token is available.